assets/bower_components/vega-elements/vega-plot.html

```diff
@@ -0,0 +1,44 @@
+<html>
+  <script src="../d3/d3.js"></script>
+  <script src="../topojson/topojson.js"></script>
+  <script src="../d3-cloud/build/d3.layout.cloud.js"></script>
+  <script src="../vega/vega.min.js"></script>
+
+  <dom-module id="vega-plot">
+    <template>
+      <div id="plot">
+          <content></content>
+      </div>
+      <br>
+      <a href="" id="pnglink" download>Save as PNG</a>
+    </template>
+    <script>
+      Polymer({
+        is: "vega-plot",
+        properties: {
+          json: {
+            type: Object,
+            notify: true,
+            observer: 'render'
+          }
+        },
+        render: function() {
+          var self = this;
+          vg.parse.spec(this.json, function (chart) {
+            chart({el:(Polymer.dom(self.$.plot)).node}).update()
+          });
+          this.async(function (){
+            this.saveAsPNG();
```

Comment:
can't this be something like `this.$.pnglink.addEventListener('click', function () { window.location = makeDataUri(); })`?

just doing this on every update seems like an overkill...